drake/solvers/Optimization.h

```diff
@@ -180,30 +284,89 @@ namespace Drake {
       v.name = name;
       v.data = Eigen::VectorXd::Zero(num_new_vars);
       v.start_index = num_vars;
-      variables.push_back(v);
       num_vars += num_new_vars;
+      variables.push_back(v);
+      variable_views.push_back(DecisionVariableView(variables.back()));
+      x_initial_guess.conservativeResize(num_vars);
+      x_initial_guess.tail(num_vars) = 0.1*Eigen::VectorXd::Random(num_vars);
 
-      return DecisionVariableView(variables.back());
+      return variable_views.back();
     }
 //    const DecisionVariable& addIntegerVariables(size_t num_new_vars, std::string name);
 //  ...
 
+    void addCost(const std::shared_ptr<Constraint>& obj) {
```

Comment:
Can we add a `Function` object as the cost instead? As `Constraint` is just `Function` with lower and upper bounds, and cost does not have bounds